bin/oref0-online.sh

```diff
@@ -1,13 +1,20 @@
 #!/bin/bash
 echo; echo Starting oref0-online.
+# if we are connected to wifi but don't have an IP, try to get one
+if iwgetid -r wlan0 | egrep -q "[A-Z]|[a-z]|[0-9]?"; then
+    if ! ip route | grep default | grep -q wlan0; then
+        echo Attempting to renew wlan0 IP
+        sudo dhclient wlan0
+    fi
+fi
 echo -n "At $(date) my local IP is: "
 ip -4 -o addr show dev wlan0 | awk '{split($4,a,"/");print a[1]}'
 ip -4 -o addr show dev bnep0 | awk '{split($4,a,"/");print a[1]}'
 echo
 echo -n "At $(date), my wifi network name is "
-printf '%s' $(iwgetid -r)
+iwgetid -r wlan0 | tr -d '\n'
 echo -n ", and my public IP is: "
-if curl --compressed -4 -s -m 15 icanhazip.com | egrep "^[12]*[0-9]*[0-9]\.[12]*[0-9]*[0-9]\.[12]*[0-9]*[0-9]\.[12]*[0-9]*[0-9]$"; then
+if curl --compressed -4 -s -m 15 icanhazip.com | awk -F , '{print $NF}' | egrep "^[12]*[0-9]*[0-9]\.[12]*[0-9]*[0-9]\.[12]*[0-9]*[0-9]\.[12]*[0-9]*[0-9]$"; then
```

Comment:
What does the `awk` statement do? I've seen false positives on restarting the wifi on my home network.
Perhaps upping `-m 15` to `-m 30` will also make it more robust, but I don't know what the response time of icanhazip.com is.